moved/src/block/in_memory.rs

```diff
@@ -92,36 +92,37 @@ impl BlockRepository for InMemoryBlockRepository {
 pub struct InMemoryBlockQueries;
 
 impl BlockQueries for InMemoryBlockQueries {
+    type Err = ();
```

Comment:
I suggest using [Infallible](https://doc.rust-lang.org/std/convert/enum.Infallible.html) when there is no possibility for error.